tools/backups2datalad/datasetter.py

```diff
@@ -557,3 +559,72 @@ async def debug_logfile(self, quiet_debug: bool) -> None:
         handler.setFormatter(fmter)
         root.addHandler(handler)
         log.info("Saving logs to %s", self.logfile)
+
+    async def get_superds_commit_message(self, superds: AsyncDataset) -> str:
+        added = []
+        modified = []
+        other = False
+        for entry in split_terminated(
+            await superds.read_git(
+                "status", "--porcelain", "--ignore-submodules=dirty", "-z", strip=False
+            ),
+            "\0",
+        ):
+            status = entry[:2]
+            path = entry[3:].rstrip("/")
+            if status[0] == "A" or status == "??":
+                added.append(path)
+            elif status[0] == "M" or status[1] == "M":
+                modified.append(path)
+            else:
+                other = True
+        added_datasets: dict[str, list[str]] = {}
+        modified_datasets: dict[str, list[str]] = {}
+        for path in added:
+            if (superds.pathobj / path / ".git").exists():
+                commits = (
+                    await AsyncDataset(superds.pathobj / path).read_git(
+                        "log", "--format=%s"
+                    )
+                ).splitlines()
+                added_datasets[path] = commits
+            else:
+                other = True
+        for path in modified:
+            if (superds.pathobj / path / ".git").exists():
+                last_commit = (
+                    (await superds.read_git("ls-tree", "-z", "HEAD", "--", path))
+                    .partition("\t")[0]
+                    .split()[2]
+                )
+                commits = (
+                    await AsyncDataset(superds.pathobj / path).read_git(
+                        "log", "--format=%s", f"{last_commit}.."
+                    )
+                ).splitlines()
+                modified_datasets[path] = commits
+            else:
+                other = True
+        msg = ""
+        if added_datasets:
+            msg += f"{len(added_datasets)} added"
+            if len(added_datasets) <= 3:
+                msg += " (" + ", ".join(sorted(added_datasets.keys())) + ")"
+        if modified_datasets:
+            if msg:
+                msg += ", "
+            msg += f"{len(modified_datasets)} updated"
+            if len(modified_datasets) <= 3:
+                msg += " (" + ", ".join(sorted(modified_datasets.keys())) + ")"
+        if other:
+            if msg:
+                msg += "; "
+            msg += "other updates"
+        if not msg:
+            msg = "unknown updates"
```

Comment:
I don't like "unknown" since they are known (committed) and could be removals etc...  may be

```suggestion
            msg = "miscellaneous changes"
```